tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -474,84 +490,100 @@ def evaluate_predictions(self, y_true, y_pred, silent=False, auxiliary_metrics=F
-                auxiliary_metrics += [f1micro_score]  # TODO: add auc?
-            # elif self.problem_type == MULTICLASS:  # multiclass metrics
-            #     auxiliary_metrics += []  # TODO: No multi-class specific metrics for now. Include top-5, top-10 accuracy here.
-
-        performance_dict = OrderedDict({metric: performance})
-        for metric_function in auxiliary_metrics:
-            if isinstance(metric_function, tuple):
-                metric_function, metric_kwargs = metric_function
+        for aux_metric in auxiliary_metrics_lst:
+            if isinstance(aux_metric, str):
+                aux_metric = get_metric(metric=aux_metric, problem_type=self.problem_type, metric_type='aux_metric')
+            if not aux_metric.needs_pred and y_pred_proba_internal is None:
+                logger.log(15, f'Skipping {aux_metric.name} because no prediction probabilities are available to score.')
+                continue
+
+            if aux_metric.name not in performance_dict:
+                if y_pred_proba_internal is not None:
+                    score = self._score_with_pred_proba(
+                        y_pred_proba_internal=y_pred_proba_internal,
+                        metric=aux_metric,
+                        **scoring_args
+                    )
+                else:
+                    score = self._score_with_pred(
+                        y_pred_internal=y_pred_internal,
+                        metric=aux_metric,
+                        **scoring_args
+                    )
+                performance_dict[aux_metric.name] = score
+
+        if self.eval_metric.name in performance_dict:
+            score_eval = performance_dict[self.eval_metric.name]
+            score_eval_flipped = self.eval_metric.convert_score_to_sklearn_val(score_eval)  # flip negative once again back to positive (so higher is no longer necessarily better)
+            if score_eval_flipped != score_eval:
+                flipped = True
             else:
-                metric_kwargs = None
-            metric_name = metric_function.__name__
-            if metric_name not in performance_dict:
-                try:  # only compute auxiliary metrics which do not error (y_pred = class-probabilities may cause some metrics to error)
-                    if metric_kwargs:
-                        performance_dict[metric_name] = metric_function(y_true, y_pred, **metric_kwargs)
-                    else:
-                        performance_dict[metric_name] = metric_function(y_true, y_pred)
-                except ValueError:
-                    pass
+                flipped = False
+            if not silent:
+                logger.log(20, f"Evaluation: {self.eval_metric.name} on test data: {score_eval}")
+                if flipped:
+                    logger.log(20, f"\tNote: Scores are always higher_is_better. This metric score can be multiplied by -1 to get the metric value.")
```

Comment:
since difference between score/value may be unclear, might clarify:
 
"metric score" -> "reported metric score"
"metric value" -> "actual {self.eval_metric.name} metric value"

Reply:
@jwmueller this sounds more confusing to me compared to the one in the log message.

Reply:
Will keep current for now, feel free to send a doc cleanup PR if you think it can be worded more clearly. I think we should get used to showing the negative values to users, as otherwise it just makes things more confusing in the long run.

Reply:
I unfortunately don't have a much better solution to this confusion. Perhaps we should also print this same message near the start of `fit()` if a flipped `eval_metric` is being used (right after where it prints "AutoGluon will gauge predictive performance using evaluation metric: ..."), because otherwise users viewing the training-progress get confused and think there is an error during training.

An alternative could be to append `negative ` (or `neg `) prefix to `eval_metric.name` for each flipped metric, but this is may get complex to deal with...